x-pack/metricbeat/module/statsd/server/data.go

```diff
@@ -162,10 +161,10 @@ func (p *metricProcessor) processSingle(m statsdMetric) error {
 		var err error
 		sampleRate, err = strconv.ParseFloat(m.sampleRate, 64)
 		if err != nil {
-			return errors.Wrapf(err, "failed to process metric `%s` sample rate `%s`", m.name, m.sampleRate)
+			return fmt.Errorf("failed to process metric `%s` sample rate `%s`: %w", m.name, m.sampleRate, err)
```

Comment:
When using errors.Wrapf, the additional context is provided by wrapping the original error. This additional context can be helpful for tracing the error back to its source and understanding the error's context, shouldn't we keep using that instead?

Reply:
[github.com/pkg/errors](https://github.com/pkg/errors) previously used is deprecated now. With the [introduction of `%w` in Go 1.13](https://go.dev/blog/go1.13-errors#:~:text=Wrapping%20errors%20with%20%25w), it is possible to wrap the error values.

Even the linters configured for the elastic org's repos are set to discourage usage of [github.com/pkg/errors](https://github.com/pkg/errors) and recommend stdlib's `error` package. So, if we do not fix it, the CI will fail.

@efd6 even wrote a detailed commit to do it using gofmt's rewrite: https://github.com/elastic/beats/commit/4ab90f9b3339c6f8a5d198ad469305dcc7bbad53 which is really handy when the changes are over a lot of files in beats. It covers almost all scenarios of errors.Wrap usage but the manual vet is also important.

Reply:
The unrelated changes in the PR are fixes for the code smells reported by the linters.